python/tank/templatekey.py

```diff
@@ -119,7 +150,7 @@ def validate(self, value):
             self._last_error = "%s Illegal value: %s is forbidden for this key." % (self, value)
             return False
 
-        if not((value is None) or (self.choices == [])):
+        if value is not None and self.choices:
```

Comment:
Could this be changed to `... and len(self.choices) > 0` ? That feels clearer to me.